src/snowflake/snowpark/async_job.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+from typing import List
+
+from snowflake.connector import SnowflakeConnection
+from snowflake.connector.cursor import ResultMetadata
+from snowflake.snowpark._internal.utils import result_set_to_iter, result_set_to_rows
+
+
+class AsyncJob:
+    def __init__(
+        self,
+        query_id: str,
+        query: str,
+        conn: SnowflakeConnection,
+        result_meta: List[ResultMetadata],
+        data_type: str = "row",
+    ) -> None:
+        self.query_id = query_id
+        self.query = query
+        self._conn = conn
+        self._cursor = self._conn.cursor()
+        self._data_type = data_type
+        self._result_meta = result_meta
+        if not conn:
+            raise ValueError("connection cannot be empty")
+        return
+
+    def is_done(self) -> bool:
+        # return a bool value to indicate whether the query is finished
+        status = self._conn.get_query_status(self.query_id)
+        is_running = self._conn.is_still_running(status)
+
+        return not is_running
+
+    def cancel(self) -> None:
+        # stop and cancel current query id
+        self._conn._cancel_query(self.query, self.query_id)
+
+    def result(self):
+        # return result of the query, in the form of a list of Row object
+        self._cursor.get_results_from_sfqid(self.query_id)
+        result_data = self._cursor.fetchall()
+        if self._data_type == "row":
```

Comment:
You can try to use Enum to have these types. Similar to https://github.com/snowflakedb/snowpark-python/blob/34a28a5f541bb9635d769381d477c2f2dea56873/src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py#L47

Reply:
sure, will do